platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/IsolatedProjectsAccessFromGroovyDslIntegrationTest.groovy

```diff
@@ -737,4 +738,74 @@ class IsolatedProjectsAccessFromGroovyDslIntegrationTest extends AbstractIsolate
             projectsConfigured(":", ":a", ":b")
         }
     }
+
+    def "fails on invoke method of unconfigured project"() {
+        given:
+        settingsFile << """
+            include(':a')
+            include(':b')
+        """
+
+        file("a/build.gradle") << """
+            def unconfiguredProject = project(':b')
+            println 'Unconfigured project value = ' + unconfiguredProject.foo()
+        """
+
+        file("b/build.gradle") << """
+            String foo(){ 'configured' }
+        """
+
+        when:
+        isolatedProjectsRun 'help', WARN_PROBLEMS_CLI_OPT
+
+        then:
+        outputContains("Unconfigured project value = null")
+        outputDoesNotContain("Unconfigured project value = configured")
+        fixture.assertStateStoredWithProblems {
+            projectsConfigured(':', ':a', ':b')
+            problem("Build file 'a/build.gradle': line 3: 'Project.evaluationDependsOn' must be used to establish a dependency between project ':b' and project ':a' evaluation")
+            problem("Build file 'a/build.gradle': line 3: Cannot access project ':b' from project ':a'")
+            problem("Build file 'a/build.gradle': line 3: Project ':b' cannot dynamically look up a method in the parent project ':'")
```

Comment:
It would be good to get rid of the last problem, because it's misleading. The real problem is that the build logic of project A is calling `.foo()` but project B's build logic is not doing anything wrong.

Reply:
Yeah, that's was my impression as well. I think that this could lead to some tests refactoring. Probably it's better to do it in separate PR